test: Fix intermittent issue in p2p_sendtxrcncl.py

maflcko opened the pull request:
Should fix https://github.com/bitcoin/bitcoin/issues/26364

I can't reproduce this, but my guess would be that `PeerNoVerack::on_version`, which sends the `wtxidrelay` message, is executed in the event loop and thus may run after the main thread sending `msg_verack`.

Also, fix another bug.

Finally, add some `assert_debug_log` to ensure the right code branch is executed (and not some random, unrelated disconnect).

Commit:
test: Fix intermittent issue in p2p_sendtxrcncl.py

Commit:
test: Check debug log as well in p2p_sendtxrcncl.py

naumenkogs commented on test/functional/p2p_sendtxrcncl.py:
```diff
@@ -162,15 +166,17 @@ def run_test(self):
             P2PInterface(), wait_for_verack=False, p2p_idx=4, connection_type="outbound-full-relay")
-        peer.send_message(sendtxrcncl_wrong_role)
-        peer.wait_for_disconnect()
+        with self.nodes[0].assert_debug_log(["sendtxrcncl received after verack"]):
```
I don't think this is the log message we should expect here. I think should be `txreconciliation protocol violation`
Does it pass because this log message happened earlier?

Naviruiz replied:
Animo

MarcoFalke replied:
`P2PInterface` sends a verack on the version message, so this will disconnect for the verack, not for another reason. Looks like another bug in the test?

naumenkogs commented on test/functional/p2p_sendtxrcncl.py:
```diff
@@ -132,12 +132,10 @@ def run_test(self):
-        peer.send_and_ping(msg_verack())
-        peer.send_message(create_sendtxrcncl_msg())
-        peer.wait_for_disconnect()
+        peer = self.nodes[0].add_p2p_connection(P2PInterface())
```
nit: worth explicitly stating `wait_for_verack=True`?

MarcoFalke replied:
I don't really like the `wait_for_verack` use in this test. I think all tests here should wait for the verack. However this is currently not possible because waiting for the verack will also send a ping-pong, which obviously doesn't work. Maybe there could be another optional arg to `add_p2p_connection` to say `skip_verack_ping_pong`?

MarcoFalke commented:
Fixed another bug in the last push

Commit:
test: Check correct disconnect reason in p2p_sendtxrcncl.py

Previously it disconnected due to "sendtxrcncl received after verack",
now it disconnects for the correct reason.

naumenkogs commented:
ACK fa3da8307baa96ea7e51cc6ba4820aca7f93ec32
